packages/lsp/src/connection.ts

```diff
@@ -0,0 +1,529 @@
+// Disclaimer/acknowledgement: Fragments are based on LspWsConnection, which is copyright of wylieconlon and contributors and ISC licenced.
+// ISC licence is, quote, "functionally equivalent to the simplified BSD and MIT licenses,
+// but without language deemed unnecessary following the Berne Convention." (Wikipedia).
+// Introduced modifications are BSD licenced, copyright JupyterLab development team.
```

Comment:
I think there is nothing left of the upstream `LspWsConnection` in this file, we should be fine using standard JupyterLab copyright header.